### DISCUSSION TITLE
output esm module syntax in server code with webpack 5

### DISCUSSION THREAD

**Discussion by dagda1** [webpack/webpack#16067] [Q&A]
I have the following webpack 5 configuration for transpiling server code which I want to transpile to esm syntax and not commonjs

```ts
const config: Configuration = merge(common, overrides, {
  name: 'server',
  target: 'node',
  experiments: {
    outputModule: true,
    topLevelAwait: true,
  },
  output: {
    path: paths.appBuild,
    filename: options.filename,
    publicPath,
    library: {
      type: 'module',
    },
    chunkFormat: 'module',
    environment: {
      const: true,
      destructuring: true,
      forOf: true,
      module: true,
      templateLiteral: true,
    },
  },
```

I have set `"type": "module"` in the package.json file.

My tsconfig.json has these compilerOptions:

```json
  "compilerOptions": {
    "baseUrl": ".",
    "target": "ESNext",
    "module": "ESNext",
```

But when I look at the output, it still has require statements:

```
const svg_namespaceObject = require("@cutting/svg");
```

**Comment by vankop** [comment]
webpack doesn't transpile your code, it is a bundler.. you need to set https://webpack.js.org/configuration/output/#outputmodule
________
also you can bundle your application to one file ( this way you don't need import/export )

**Reply by dagda1** [reply]
if you bundle it to one file then you would need to bundle all of node_modules
